include/seqan/parallel/parallel_lock.h

```diff
@@ -213,14 +213,26 @@ struct ScopedWriteLock<TLock, Serial>
 inline void
 yieldProcessor()
 {
-#if defined(__arm__) || defined(_M_ARM)
-    asm volatile ("yield" ::: "memory");
-#elif defined(STDLIB_VS)
+#if defined(STDLIB_VS)  // Visual Studio - all platforms.
     YieldProcessor();
-#elif defined(__SSE2__)
-    _mm_pause();
+#elif defined(__arm__) || defined(__aarch64__)  // ARM.
+    __asm__ __volatile__ ("yield" ::: "memory");
+#elif defined(__sparc) || defined(__sparc__)  // SPARC
+#if defined(__SUNPRO_C)
+    __asm __volatile__ ("rd %%ccr, %%g0\n\t"
+                        "rd %%ccr, %%g0\n\t"
+                        "rd %%ccr, %%g0")
 #else
-    __asm__ __volatile__("rep; nop" : : );
+    __asm __volatile__ ("rd %ccr, %g0\n\t"
+                        "rd %ccr, %g0\n\t"
+                        "rd %ccr, %g0")
+#endif  // defined(__SUNPRO_C)
+#elif defined(__ppc__) || defined(__ppc64__) || defined(_ARCH_PPC)  || defined(_ARCH_PPC64)  // PowerPC
```

Comment:
Can we change this to:
```c++
#elif defined(__powerpc__) || defined(__ppc__) || defined(__PPC__) // PowerPC
```
first is generic, second is for macOS, third is for OpenBSD
Taken from:
http://nadeausoftware.com/articles/2012/02/c_c_tip_how_detect_processor_type_using_compiler_predefined_macros